SupplyQueryDemo/SupplyQueryDemo/Program.cs

```diff
@@ -0,0 +1,51 @@
+﻿using SupplyQueryDemo;
+
+const string Query = @"
+query Search($mpn: String!) {
+  supSearchMpn(q: $mpn, limit: 2) {
+    results {
+      part {
+        mpn
+        shortDescription
+        manufacturer {
+          name
+        }
+      }
+    }
+  }
+}";
+
+while (true)
+{
+    // prompt for an MPN to search
+    Console.Write("Search MPN: ");
+    var mpn = Console.ReadLine();
+    if (string.IsNullOrEmpty(mpn))
+        return;
+
+    using HttpClient supplyClient = await SupplyClient.GetClientAsync();
+
+    // run the query
+    Request request = new()
+    {
+        Query = Query,
+        Variables = new Dictionary<string, object> { { "mpn", mpn } }
+    };
+
+    Response result = await supplyClient.RunQueryAsync(request);
+
+    if (result.Data?.SupSearchMpn?.Results == null)
+    {
+        continue;
```

Comment:
I'd be tempted to output a line here: `($"Sorry no part found for MPN search: {mpn}")`

Reply:
The other option is to handle errors as returned from the GraphQL response, e.g. something like this...
```
    {
        [JsonProperty("data")]
        public Data Data { get; set; }

        [JsonProperty("errors")]
        public List<Error> Errors { get; set; }
    }

    public partial class Error
    {
        [JsonProperty("message")]
        public string Message { get; set; }
    }
```    
... but that probably adds complexity for what we want here.